testing/snapshot.ts

```diff
@@ -1,27 +1,66 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
-import { fromFileUrl, join, parse, toFileUrl } from "../path/mod.ts";
+import {
+  basename,
+  fromFileUrl,
+  join,
+  parse,
+  resolve,
+  toFileUrl,
+} from "../path/mod.ts";
 import { ensureFile, ensureFileSync } from "../fs/mod.ts";
 import { bold, green, red } from "../fmt/colors.ts";
-import { AssertionError, equal } from "./asserts.ts";
+import { assert, AssertionError, equal } from "./asserts.ts";
 import { buildMessage, diff, diffstr } from "./_diff.ts";
 
 const CAN_NOT_DISPLAY = "[Cannot display]";
 const SNAPSHOT_DIR = "__snapshots__";
+const SNAPSHOT_EXT = "snap";
 
-type AssertSnapshotContext = {
-  snapshotFileUrl: URL | null;
-  teardownRegistered: boolean;
-  currentSnapshot: Map<string, string> | null;
-  updatedSnapshot: Map<string, string>;
-  snapshotCounts: Map<string, number>;
-  snapshotUpdatedCount: number;
+export type SnapshotMode = "assert" | "update";
+
+export type SnapshotOptions<T = unknown> = {
+  /**
+   * Snapshot output directory. Snapshot files will be written to this directory.
+   * This can be relative to the test directory or an absolute path.
+   */
+  dir?: string;
+  /**
+   * Snapshot mode. Defaults to `assert`, unless the `-u` or `--update` flag is
+   * passed, in which case this will be set to `update`. This option takes higher
+   * priority than the update flag. If the `--update` flag is passed, it will be
+   * ignored if the `mode` option is set.
+   */
+  mode?: SnapshotMode;
+  /**
+   * Failure message to log when the assertion fails. Specifying this option will
+   * cause the diff not to be logged.
+   */
+  msg?: string;
+  /**
+   * Name of the snapshot to use in the snapshot file.
+   */
+  name?: string;
+  /**
+   * Snapshot output path. The shapshot will be written to this file. This can be
+   * a path relative to the test directory or an absolute path.
```

Comment:
Please add a comment that this option will override `dir` if the path is absolute

Reply:
Done 👍